ambry-cloud/src/main/java/com.github.ambry.cloud/azure/AzureCloudDestination.java

```diff
@@ -69,6 +73,8 @@
   private final RequestOptions defaultRequestOptions = new RequestOptions();
   private final OperationContext blobOpContext = new OperationContext();
   private final AzureMetrics azureMetrics;
+  // Containers known to exist in the storage account
+  private final Set<String> knownContainers = Collections.synchronizedSet(new HashSet<>());
```

Comment:
Create thread safe hash set from ConcurrentHashMap for better concurrency?

ConcurrentHashMap<String, Integer> map = new ConcurrentHashMap<>(); 
knownContainers = map.newKeySet();

Reply:
Can you actually modify a key set that is backed by a map?

Reply:
Never mind, you said newKeySet(), not keySet().